services/outputhost/messagecache.go

```diff
@@ -979,12 +990,20 @@ func (msgCache *cgMsgCache) updateConn(connID int, event msgEvent) {
 
 func (msgCache *cgMsgCache) refreshCgConfig(oldOutstandingMessages int32) {
 	outstandingMsgs := oldOutstandingMessages
+	redeliveryIntervalInMs := msgCache.redeliveryIntervalInMs
+
 	cfg, err := msgCache.cgCache.getDynamicCgConfig()
 	if err == nil {
 		outstandingMsgs = msgCache.cgCache.getMessageCacheSize(cfg, oldOutstandingMessages)
+		redeliveryIntervalInMs = cfg.RedeliveryIntervalInMs
 	}
 
 	msgCache.maxOutstandingMsgs = outstandingMsgs
+
+	if redeliveryIntervalInMs != msgCache.redeliveryIntervalInMs {
+		msgCache.redeliveryTicker = time.NewTicker(time.Duration(redeliveryIntervalInMs) * time.Millisecond)
```

Comment:
before creating a new ticker, you should "stop the ticker to release associated resources" (https://golang.org/pkg/time/#NewTicker).

Reply:
Nice!